specification/network/resource-manager/Microsoft.Network/2017-09-01/applicationSecurityGroup.json

```diff
@@ -0,0 +1,370 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationSecurityGroups/{applicationSecurityGroupName}/referencingSecurityRules/$ref": {
+      "get": {
+        "tags": [
+          "ApplicationSecurityGroups"
+        ],
+        "operationId": "ApplicationSecurityGroups_ListReferencingSecurityRules",
+        "description": "This operation returns the security rules that reference the specified application security group resource.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "applicationSecurityGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application security group."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the security rules that reference the specified application security group resource.",
+            "schema": {
+              "$ref": "./network.json#/definitions/ResourceReferenceResultList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List all security rules that reference application security group": { "$ref": "./examples/ApplicationSecurityGroupListReferencingSecurityRules.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationSecurityGroup": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationSecurityGroupPropertiesFormat",
+          "description": "Properties of the application security group."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "An application security group in a resource group."
+    },
+    "ApplicationSecurityGroupPropertiesFormat": {
+      "properties": {
+        "resourceGuid": {
+          "type": "string",
+          "description": "The resource GUID property of the application security group resource."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The provisioning state of the application security group resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
```

Comment:
> Possible values are: 'Updating', 'Deleting', and 'Failed'

What about Success or cancelled ?

Reply:
make this an enum

Reply:
Fixed. I checked the code and "Cancelled" does not exist.